Sources/App/Scenes/WindowSizeObserver.swift

```diff
@@ -1,16 +1,17 @@
 import Foundation
+import Shared
 import UIKit
 
 final class WindowSizeObserver: NSObject {
     @objc private(set) var observedScene: UIWindowScene?
-    private var observation: NSKeyValueObservation?
+    private weak var observation: NSKeyValueObservation?
```

Comment:
this feels a little weird. do we want to stop observation more concretely? does this token automatically invalidate on deinit? iirc it does not but just wanna double check

Reply:
I have updated the logic to explicit invalidate the observation, I noticed that the previous approach was not even saving the window size anymore so it was good to now have merged it yet